addon/components/ce/content-editable.js

```diff
@@ -318,19 +322,51 @@ export default class ContentEditable extends Component {
     }
   }
 
-  /**
-   * cut isn't allowed at the moment
-   */
   @action
   cut(event) {
     event.preventDefault();
+
+    const htmlExportWriter = new HTMLExportWriter(this.rawEditor.model);
+    const modelNodes = this.rawEditor.executeCommand("delete-selection");
+
+    // Filter out model nodes that are text related
+    const filter = toFilterSkipFalse(node => {
+      return ModelNode.isModelText(node) || (ModelNode.isModelElement(node) && node.type === "br");
+    });
+
+    let htmlString = "";
+    let htmlTextString = "";
+    for (const modelNode of modelNodes) {
+      if (ModelNode.isModelElement(modelNode)) {
+        modelNode.parent = null;
+        const range = ModelRange.fromAroundNode(modelNode);
+        const treeWalker = new ModelTreeWalker({filter, range});
+
+        for (const node of treeWalker) {
+          htmlTextString += ModelNode.isModelText(node)
+            ? node.content
+            : "\n";
+        }
+      } else {
+        htmlTextString += modelNode.content;
+      }
+
+      const node = htmlExportWriter.write(modelNode);
+      htmlString += node instanceof HTMLElement
+        ? node.outerHTML
+        : node.textContent;
+    }
+
+    const clipboardData = event.clipboardData || window.clipboardData;
+    clipboardData.setData("text/plain", htmlTextString);
+    clipboardData.setData("text/html", htmlString);
   }
 
   /**
    * copy is relegated to the browser for now
    */
   @action
-  copy( /* event */) {
+  copy(/* event */) {
```

Comment:
I realized earlier this week that we should probably provide a copy handler as well and that the behaviour would be very similar to the cut command

Reply:
I think it will indeed be very similar to the cut command and we can reuse most of the code from it (except for the part that removes the selected nodes). Also, the split that gets performed when doing a cut should not be performed on the actual HTML. But since we are using a virtual DOM, this shouldn't be a big problem.